### DISCUSSION TITLE
Discussions API: Allow specifying order in DiscussionCommentConnection

### DISCUSSION THREAD

**Discussion by laymonage** [community/community#15432] [Discussions]
The `DiscussionCommentConnection`'s signature only allows `after`, `before`, `first`, and `last` fields to be specified. This means that we can't specify the order of the comments to be returned by the API. It would be nice to have the same ordering options as the web interface:

<img width="242" alt="image" src="https://user-images.githubusercontent.com/6379424/164873252-9819f928-0143-4482-8ad3-1b2252cac5c4.png">

I understand that both the `comments` and `replies` use the same `DiscussionCommentConnection` interface, so it might not make sense to have the "Top" ordering on replies. However, being able to specify the order at least by Oldest/Newest would already be a big improvement.

**Comment by AustenLamacraft** [comment]
So what order is implied by `first` and `last`? I assumed it was by `createdAt`, or `updatedAt`.

**Reply by laymonage** [reply]
It's ordered from oldest to newest by `createdAt`.

**Reply by AustenLamacraft** [reply]
Do you know how you'd get the latest activity on a discussion? You could get the last comment, but that would miss the possibility that there was a more recent reply to an older comment. The alternative is to get all the comments and to look through them, though that seems a bit wasteful.

**Reply by laymonage** [reply]
No, I don't think there's a way to do that currently. Best you can do is probably fetch all the top-level comments and the last 1 reply of each comment, then sort the results on your own.

**Comment by Bestra** [comment]
I've written up an issue for this in our backlog. We have some internal plumbing available that I hope will make this one pretty straightforward to implement.

**Reply by ryanwwest** [reply]
Very much looking forward to this for sorting by Top comments via API.

**Reply by Bestra** [reply]
I'll see if I can get this prioritized, time flew over the holidays 😅

**Comment by laymonage** [comment]
Bump.
